packages/near-fast-auth-signer/src/lib/controller.ts

```diff
@@ -145,8 +155,41 @@ class FastAuthController {
   }
 
   async signDelegateAction({ receiverId, actions, signerId }) {
-    this.assertValidSigner(signerId);
+    const isSigned = await this.isSignedIn();
+    if (this.accountId !== signerId) {
+      this.setAccountId(signerId);
+      this.localStore = new keyStores.BrowserLocalStorageKeyStore();
+    }
```

Comment:
What is the purpose of this code block? AFAICT the only time we override the `accountId` is when there is no `accountId` in the controller and in that case we derive it from the oidc token and set it

Reply:
I have add them in because in some cases,
controller is initialised with:
https://github.com/near/fast-auth-signer/blob/1cd623db107e12109de7077aa7cf0c7dd009f164/packages/near-fast-auth-signer/src/App.tsx#L22-L25
where accountId is wrong.